### PR TITLE
DOCSP-36049 AI and Data Usage Page

### PR DESCRIPTION
**Pull request by ianf-mongodb**
## DESCRIPTION

- Adding a `How Query Converter Uses Your Data` page to the `Code Generation` drawer explaining details on how   user's metadata are used WRT query converter.
- Also includes an update to the enable query converter page, for optionally disabling telemetry.
- Removes all references to Query Converter being a private preview feature.

[Copy Draft](https://docs.google.com/document/d/1iHhXescsma0lkrSUQ49niehpJP1unehc4JkzgSuK5Xw/edit#heading=h.nspknerwmj97)

Dochub link: https://dochub.mongodb.org/core/migrator-ai-data-usage

[Figma](https://www.figma.com/proto/oSw5TTXpRXghvId3ZF5JoE/Query-Converter%3A-Public-Release?type=design&node-id=29-48270&t=wgJGeDA0kTprg8UI-0&scaling=min-zoom&page-id=0%3A1&starting-point-node-id=29%3A48270)

## STAGING

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-36049/code-generation/query-converter/ai-and-data-usage-information/

https://preview-mongodbianfmongodb.gatsbyjs.io/docs-relational-migrator/DOCSP-36049/code-generation/query-converter/enable-query-converter/

## JIRA

https://jira.mongodb.org/browse/DOCSP-36049

## BUILD LOG

https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65d8c5b17111165982a2d9d8

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/code-generation/query-converter/enable-query-converter.txt [hunk 1]
```diff
@@ -36,11 +36,27 @@ Steps
 
    .. step:: Read the user agreement
 
+   .. step:: (Optional) Disable telemetry
```
Comment (tomhollander) [inline source/code-generation/query-converter/enable-query-converter.txt:39]: Can we make the heading "Review telemetry preferences" or similar? We don't want to encourage disabling, so I'd like the language to be softer.

**Review by tomhollander — CHANGES_REQUESTED** [review]
One minor copy change, plus we need to remove the references to the private preview.

File: source/code-generation/query-converter/enable-query-converter.txt [hunk 2]
```diff
@@ -36,11 +28,27 @@ Steps
...
+      .. note::
+
+         You can also disable telemetry data by toggling the 
+         :guilabel:`Allow queries to be stored temporarily` 
+         switch in global project settings.
+
    .. step:: Enable the query converter feature
 
-      Click :guilabel:`Use Query Converter`. 
+      Click :guilabel:`Start Using Query Converter`.
```
Comment (ianf-mongodb) [inline source/code-generation/query-converter/enable-query-converter.txt:48]: I think based on your screenshots yesterday this button text has changed in the next release:
Screenshot 2024-02-22 at 10 12 05 AM

**Review by ianf-mongodb — COMMENTED** [review]
Ready for another look @tomhollander

File: source/code-generation/query-converter/how-query-converter-uses-your-data.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+How Query Converter Uses Your Data
+==================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Query Converter currently uses the `Azure OpenAI Service
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/how-query-converter-uses-your-data.txt:13]:
```suggestion
Query Converter now uses the `Azure OpenAI Service
```

File: source/code-generation/query-converter/how-query-converter-uses-your-data.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  and data types.
+
+The information that is sent is not shared with any other third 
+parties or stored by the AI provider. Relational Migrator does not send 
+database connection strings, credentials, or any raw data from your 
+databases.
+
+By default, your original query text and the AI-converted query 
+telemetry data is stored by MongoDB for up to one year to help us 
+provide support to customers and improve the service. If you do not want
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/how-query-converter-uses-your-data.txt:36]: [minimalism]
```suggestion
provide support and improve the service. If you do not want
```

File: source/code-generation/query-converter/how-query-converter-uses-your-data.txt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+your data to be stored, you have the option to opt out:
+
+- You can disable the :guilabel:`Allow queries to be stored temporarily` 
+  toggle switch in global project settings.
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/how-query-converter-uses-your-data.txt:40]:
```suggestion
  toggle switch in the global project settings.
```

File: source/code-generation/query-converter/how-query-converter-uses-your-data.txt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+- You can uncheck the :guilabel:`Allow MongoDB to temporarily store the 
+  source and converted queries to import the product feature` setting 
+  when you enable query converter.
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/how-query-converter-uses-your-data.txt:43]:
```suggestion
  when you enable the query converter.
```

File: source/code-generation/query-converter/how-query-converter-uses-your-data.txt [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Query Converter currently uses the `Azure OpenAI Service
+<https://azure.microsoft.com/en-us/products/ai-services/openai-service>`__ 
+hosted by Microsoft. 
+
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/how-query-converter-uses-your-data.txt:16]: [Suggestion]

I know this page isn't the traditional concept page but it is still covering mostly conceptual Query Converter info, so maybe we can add a "Behavior" heading here?

File: source/code-generation/query-converter/how-query-converter-uses-your-data.txt [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+provide support to customers and improve the service. If you do not want
+your data to be stored, you have the option to opt out:
+
+- You can disable the :guilabel:`Allow queries to be stored temporarily` 
+  toggle switch in global project settings.
+- You can uncheck the :guilabel:`Allow MongoDB to temporarily store the 
+  source and converted queries to import the product feature` setting 
+  when you enable query converter.
+
+Disable Query Converter
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/how-query-converter-uses-your-data.txt:45]: [Question/Suggestion]

So technically we can't properly disable this feature but we can simply not use it/enable it? If so, maybe we can make this heading the "Details" heading from concept page prototype. I think with the current title it does come across like you can turn it on and off.

File: source/code-generation/query-converter/enable-query-converter.txt [hunk 9]
```diff
@@ -36,11 +28,27 @@ Steps
...
+
+         You can also disable telemetry data by toggling the 
+         :guilabel:`Allow queries to be stored temporarily` 
+         switch in global project settings.
+
    .. step:: Enable the query converter feature
 
-      Click :guilabel:`Use Query Converter`. 
+      Click :guilabel:`Start Using Query Converter`. 
       The :guilabel:`Query Converter` pane displays.
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/enable-query-converter.txt:49]: [Question]

I am just a little confused on the last sentence, was it suppose to be something along the lines of "The Query Converter pane is displayed."?

**Review by jocelyn-mendez1 — COMMENTED** [review]
Thanks for the updates @ianf-mongodb! I left some suggestions and questions so please let me know if I can clarify any of that.

File: source/code-generation/query-converter/enable-query-converter.txt (hunk 9 above)
Comment (ianf-mongodb) [reply source/code-generation/query-converter/enable-query-converter.txt]: Yes, Updated.

**Review by ianf-mongodb — COMMENTED** [review]
Thanks @jocelyn-mendez1 this is ready for another look!

**Review by tomhollander — APPROVED** [review]

**Comment by tomhollander** [comment]
Looks good! Note we will need to coordinate the publishing of this content to coincide with the next release (likely next week or the week after)

File: source/code-generation/query-converter/ai-and-data-usage-information.txt [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+How Your Data is Used
+---------------------
+
+When you use Query Converter, the following information is sent to 
+MongoDB's backend and the third party AI provider:
+
+- The full text of your query or stored procedure.
+- The definition of any relational database tables used in the query.
+  These definitions include the table names, column names, data types,
+  and any primary or foreign keys relationships.
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/ai-and-data-usage-information.txt:34]:
```suggestion
  and any primary or foreign key relationships.
```

File: source/code-generation/query-converter/ai-and-data-usage-information.txt [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+- You can disable the :guilabel:`Allow queries to be stored temporarily` 
+  toggle switch in the global project settings.
+- You can uncheck the :guilabel:`Allow MongoDB to temporarily store the 
+  source and converted queries to import the product feature` setting 
+  when you enable the query converter.
+
+Disable Query Converter
+-----------------------
+
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/ai-and-data-usage-information.txt:58]: [Question]

Do you think we could link to the Enable Query Converter page here?

Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/ai-and-data-usage-information.txt:58]: [Question]

Do you think we could link to the Enable Query Converter page here?

File: source/code-generation/query-converter/enable-query-converter.txt [hunk 12]
```diff
@@ -36,10 +28,25 @@ Steps
...
+   .. step:: (Optional) Review telemetry preferences
+
+      Telemetry data about the source and converted queries are 
+      temporarily stored by MongoDB to improve the query converter 
+      feature. You can disable sending telemetry data by unchecking the 
+      :guilabel:`Allow MongoDB to temporarily store source and coverted 
+      queries to improve the product feature`
+      check box.
+
+      .. note::
```
Comment (jocelyn-mendez1) [inline source/code-generation/query-converter/enable-query-converter.txt:40]: [Question/Suggestion]

Currently having one disabling option in the note admonition makes it seem like that one has a different level of importance from the other. I really like the way you wrote this in the AI and Data Usage page with the listed bullets. Do you think we could do that here?

**Review by jocelyn-mendez1 — COMMENTED** [review]
Hi @ianf-mongodb, thanks for the updates this is looking really good! I left some small suggestions so let me know if I can clarify any of that.

**Review by jocelyn-mendez1 — COMMENTED** [review]
Hi @ianf-mongodb, thanks for the updates this is looking really good! I left some small suggestions so let me know if I can clarify any of that.

File: source/code-generation/query-converter/ai-and-data-usage-information.txt (hunk 11 above)
Comment (ianf-mongodb) [reply source/code-generation/query-converter/ai-and-data-usage-information.txt:54]: Yes definitely, done!

File: source/code-generation/query-converter/enable-query-converter.txt (hunk 12 above)
Comment (ianf-mongodb) [reply source/code-generation/query-converter/enable-query-converter.txt]: Yes totally, I converted that piece to include and those bullets are now shared on each page.

**Review by ianf-mongodb — COMMENTED** [review]
Thanks @jocelyn-mendez1 I have updated and this is ready for another look.

**Review by jocelyn-mendez1 — APPROVED** [review]
LGTM!

**Review by themantissa — APPROVED** [review]
Two small adjustments to match copy agreed to with legal.

**Pull request merged**